lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Library.java

```diff
@@ -32,6 +33,18 @@ public String getId() {
     return this.id;
   }
 
+  public void addMediaItem(MediaItem mediaItem) {
```

Comment:
Almost there.  With these methods, you shouldn't need to have methods `addBook`, `removeBook`, and `hasBook`.  You should be able to rename those methods, swap out `Book` for `MediaItem` and still see the tests pass.